flake.nix

```diff
@@ -64,7 +64,7 @@
               pkgs.go-licenses
               pkgs.go-task
               pkgs.go-tools
-              pkgs.go_1_23
+              pkgs.go_1_24
```

Comment:
I'm not opposed to upgrading to 1.24 but let's do so in a distinct commit/PR so we can easily. Did rpk update to go 1.24 which requires us to upgrade as well?